src/base/rss/rss_feed.cpp

```diff
@@ -207,13 +207,14 @@ void Feed::handleParsingFinished(const RSS::Private::ParsingResult &result)
         m_lastBuildDate = result.lastBuildDate;
 
         foreach (const QVariantHash &varHash, result.articles) {
-            auto article = Article::fromVariantHash(this, varHash);
-            if (article) {
+            try {
+                auto article = new Article(this, varHash);
                 if (!addArticle(article))
                     delete article;
                 else
                     m_dirty = true;
             }
+            catch (...) {}
```

Comment:
I just rewritten old code using exceptions instead of static "create" method.
This behavior is expected. When we have incorrect data (i.e. when we haven't article id or some other id-like field) we cannot construct valid Article object. We just ignore it.